monix-execution/shared/src/main/scala/monix/execution/misc/Local.scala

```diff
@@ -78,12 +95,18 @@ object Local {
   def bindClear[R](f: => R): R =
     macro Macros.localLetClear
 
-  /** If `b` evaluates to `true`, execute a block of code using a current
+  /** If `b` evaluates to `true`, execute a synchronous block of code using a current
     * state of `Local.Context` and restore the current state when complete.
     */
   def bindCurrentIf[R](b: => Boolean)(f: => R): R =
     macro Macros.localLetCurrentIf
 
+  /** If `b` evaluates to `true`, execute an asynchronous block of code using a current
+    * state of `Local.Context` and restore the current state when complete.
+    */
+  def bindCurrentAsyncIf[R](b: => Boolean)(f: => Future[R]): Future[R] =
```

Comment:
couldn't just overload because of `(b: => Boolean)` I suppose

Reply:
There's actually no point in having `b` being by-name here. It will always be strictly evaluated.

Reply:
Good point, I'll change it although it is still ambigious reference to overloeaded definition if we prefer overload.